src/averageAges.js

```diff
@@ -14,12 +14,28 @@
  *
  * @return {number}
  */
+
+function getAverageAge(humans) {
+  const averageAges = humans.reduce(
+    (sum, { born, died }) => sum + (died - born),
+    0);
+
+  return averageAges / humans.length;
+}
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const menFilterFunction = (person) => {
+    const isMan = person.sex === 'm';
+    const isCentury = Math.ceil(person.died / 100) === century;
```

Comment:
This calculation will happen every iteration. Why not make it a part of decision-making?
```suggestion
    const isFromCentury = century ? Math.ceil(person.died / 100) === century : true;
```